src/App.js

```diff
@@ -1,13 +1,162 @@
 import React, { Component } from 'react';
+import ContactList from './ContactList';
+import SearchBar from './SearchBar';
+import ContactForm from './ContactForm';
+import axios from 'axios';
 
 class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      searchText: '',
+      contacts: [],
+      addedContacts: [],
+    };
+  }
+
+  handleSearchBarChange(event) {
+    this.setState({
+      searchText: event.target.value
+    });
+  }
+
+  componentDidMount() {
+    axios.get('http://localhost:4000/contacts')
+      .then(resp => {
+        this.setState({
+          contacts: resp.data
+        });
+      })
+      .catch(err => {
+        console.log(`Error! ${err}`)
+      });
+  }
+
+  handleChange(event) {
+    this.setState({
+      searchText: event.target.value
+    });
+  }
+
+  componentWillMount() {
+    console.log('componentWillMount');
+  }
+
+  getFilteredContacts() {
+
+    const term = this.state.searchText.trim().toLowerCase();
+    const contacts = this.state.contacts;
+
+    if (!term) {
+      return contacts;
+    }
+
+    return contacts.filter(contact => {
+      return contact.name.toLowerCase().indexOf(term) >= 0;
+    });
+  }
+
+  handleAddContact(attributes) {
+    axios.post('http://localhost:4000/contacts', attributes)
+      .then(resp => {
+        this.setState({
+          contacts: [...this.state.contacts, resp.data]
+        });
+      })
+      .catch(err => console.log(err));
+  }
+
+  addContact(contact) {
+    const curContacts = [...this.state.contacts];
+    const curContact = curContacts[contact - 1];
+    const added = [...this.state.addedContacts];
+    const newContact = {
+      _id: curContact._id + 10,
+      name: curContact.name,
+      occupation: curContact.occupation,
+      avatar: curContact.avatar,
+      active: '',
+      buttonType: 'Remove'
+    };
+    added.push(newContact);
+    this.setState({
+      addedContacts: added,
+    });
+
+    this.popContact(contact);
+  }
+
+  reset() {
+    const curContacts = [...this.state.contacts];
+
+    for (let i = 0; i < curContacts.length; i++) {
+      curContacts[i].active = '';
+    }
+
+    this.setState({
+      contacts: curContacts,
+      addedContacts: [],
+    });
+
+  }
+
+  popContact(contact) {
+
+    const visContacts = [...this.state.contacts];
+    visContacts[contact - 1].active = 'none';
```

Comment:
Use boolean data type for boolean values. You are mutating the contact object here. (Only the array was cloned above) Better to clone the contact object and change the value in the cloned object.